Sources/NIOHTTP2/HTTP2CommonInboundStreamMultiplexer.swift

```diff
@@ -321,3 +362,80 @@ extension HTTP2CommonInboundStreamMultiplexer {
         }
     }
 }
+
+extension HTTP2CommonInboundStreamMultiplexer {
+    func setStreamChannels(_ streamChannels: any ContinuationSanitizer) {
+        self.streamChannels = streamChannels
+    }
+}
+
+/// `ContinuationSanitizer` is used to type-erase generic async-sequence-like objects. This is so that they may be held
+/// by the `HTTP2ChannelHandler` without causing it to become generic itself.
+internal protocol ContinuationSanitizer {
```

Comment:
I think we are type-erasing, we're erasing the generic `Output` type are we not?

Reply:
Not really. In this type, we are specialising the generic type of `AsyncThrowingStream<Element>.Continuation` to be `Channel`. We do this so that we can avoid spelling out the `Output` type inside the multiplexer/handler. While it has similar properties has type erasure I wouldn't really call it this here :D
Overall I think we should just change the name of the protocol `ContinuationSanitizer` seems very vague and the type does something different that I expected from the name.